pkg/server/container_create.go

```diff
@@ -92,7 +92,11 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 	// Reserve the container name to avoid concurrent `CreateContainer` request creating
 	// the same container.
 	id := util.GenerateID()
-	name := makeContainerName(config.GetMetadata(), sandboxConfig.GetMetadata())
+	metadata := config.GetMetadata()
+	if metadata == nil {
+		return nil, errors.New("GetMetadata failed")
```

Comment:
nit: Go error should start with a lower character, e.g. `errors.New("failed to get metadata")`

Reply:
/s/GetMetadata failed/container config must include metadata/

add a TODO to make container create work using image name if metadata name is not set..